addons/source-python/packages/source-python/core/__init__.py

```diff
@@ -101,6 +107,36 @@ def _unload_instance(self):
             'have its own implementation of an _unload_instance method.')
 
 
+class WeakAutoUnload(AutoUnload):
+    """Subclass of AutoUnload used to unload specific instances when all
+    references to the instance are deleted.
+    """
+
+    def new(cls, *args, **kwargs):
+        self = super().__new__(cls)
+        self.___unloaded = False
```

Comment:
@Ayuto Those aren't super-private attributes, they're to avoid name collision in subclasses like @jsza said. [Here](https://www.youtube.com/watch?v=HTLu2DFOdTg&t=33m8s)'s a good talk about it from Raymond Hettinger.